app/models/ems_refresh/save_inventory_network.rb

```diff
@@ -237,12 +237,12 @@ def save_load_balancers_inventory(ems, hashes, target = nil)
     store_ids_for_new_records(ems.load_balancers, hashes, :ems_ref)
   end
 
-  def save_network_ports_inventory(ems, hashes, target = nil)
+  def save_network_ports_inventory(ems, hashes, target = nil, mode = :refresh)
     target = ems if target.nil?
 
     ems.network_ports.reset
     deletes = if target == ems
-                :use_association
+                ems.network_ports.where(:source => mode).dup
```

Comment:
hm, we should probably model this with association, not sure if we can do it now though